### PR TITLE
dev-perl/Devel-Cover: version bump

### PR DESCRIPTION
**Pull request by bor**

### CODE REVIEW

**Comment by monsieurp** [comment]
@gentoo/perl @kentfredric

File: dev-perl/Devel-Cover/Devel-Cover-1.230.0.ebuild [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+# Distributed under the terms of the GNU General Public License v2
+# $Id$
+
+EAPI=6
+
+DIST_AUTHOR=PJCJ
+DIST_VERSION=1.23
+inherit perl-module
+
+DESCRIPTION='Code coverage metrics for Perl'
```
Comment (kentfredric) [inline dev-perl/Devel-Cover/Devel-Cover-1.230.0.ebuild]: Please don't use single quotes unless necessary.

Comment (bor) [inline dev-perl/Devel-Cover/Devel-Cover-1.230.0.ebuild]: sorry, were was copy-paste :)
will fix

Comment (bor) [inline dev-perl/Devel-Cover/Devel-Cover-1.230.0.ebuild]: fixed

**Comment by kentfredric** [comment]
Note there is much room for improvement on this dist, with *many* undeclared externalities that can be triggered in codepaths.

Static Analysis of just META.json turned up this:
```
RDEPEND:
 !minimal? (
     Browser::Open → 0
     Capture::Tiny → 0
     Class::XSAccessor → 0
     virtual/perl-JSON-PP → 0
     Moo → 0
     PPI::HTML → 1.07
     Parallel::Iterator → 0
     Pod::Coverage → 0.06
     Pod::Coverage::CountParents → 0
     Template → 2.00
     Test::Differences → 0
     namespace::clean → 0
     dev-perl/perltidy → 20060719
 )
 virtual/perl-Digest-MD5 → 0
 virtual/perl-Storable → 0
DEPEND:
  ${RDEPEND}
 virtual/perl-ExtUtils-MakeMaker → 0
 test? (
     virtual/perl-Test-Simple → 0
 )
```

And many of these *should* be installed by default because they provide usable functionality.

**Comment by bor** [comment]
AFAIU these are an optional dependencies, and module should works fine without.
And it does as for me ;)

**Comment by kentfredric** [comment]
Yep. But that's why "minimal" is a thing.

"Minimal" is the gentoo USE flag that says "Ok, I don't mind if some features are missing/broken, you can skip these dependencies".

But we're supposed to make *all* the features work out of the box, and where its not possible/desired, we should fence them off with useflags.

**Comment by kentfredric** [comment]
BTW: "Works fine for me" is just because 

1. You probably have the dependencies anyway
2. Or you're not using the feature that requires them.

For instance, you don't have dev-perl/Browser-Open, so I'm sure that `cover -launch` will get you a nice big "die" instead of opening your browser.

**Comment by kentfredric** [comment]
Perhaps you can excuse doing it, just I'll have to todo-this at some time and do that stuff proper ( which requires new deps ):

TL;DR, we're currently shipping incomplete software. Maintaining that status quo is easy.

But we should aim to do better in the long term.

**Comment by bor** [comment]
I'm agree with you :)

About current case: for me it good enough, that's why I'm not going to improve it better, sorry.
And about
```
$ cover -launch
...
HTML output written to ...
done.
Devel::Cover: -launch requires Browser::Open
```
IMHO it is a proper handling ;)

**Comment by monsieurp** [comment]
@kentfredric your points are valid. yet this problem should be sorted out in a different bug report. I'll merge this contribution though and I'll file a bug for this package to add the missing deps. Thanks for your work @bor!
